robottelo/hosts.py

```diff
@@ -1848,8 +1848,30 @@ def cli(self):
     @contextmanager
     def omit_credentials(self):
         self.omitting_credentials = True
+        # if CLI is already created
+        if self._cli._configured:
+            for file in Path('robottelo/cli/').iterdir():
+                if file.suffix == '.py' and not file.name.startswith('_'):
+                    cli_module = importlib.import_module(f'robottelo.cli.{file.stem}')
+                    for name, obj in cli_module.__dict__.items():
+                        try:
+                            if Base in obj.mro():
+                                getattr(self._cli, name).omitting_credentials = True
+                        except AttributeError:
+                            # not everything has an mro method, we don't care about them
+                            pass
```

Comment:
Initially, @jameerpathan111 and I had some concerns about code repetition. But looking at it further, we concluded that it could be possibly remade into the following:
```suggestion
            for name, obj in self._cli.__dict__.items():
                with contextlib.suppress(AttributeError):  # not everything has an mro method, we don't care about them
                    if Base in obj.mro():
                        getattr(self._cli, name).omitting_credentials = True
```

With this change, it does not seem that much repetitive. :smiley:  We are not sure if it would work, though. But if the `omit_credentials` property assumes `self._cli._configured`, then we can rely on the classes being already populated there instead of going over `robottelo/cli/`.

Reply:
I think cycling through `robottelo/cli` ensures we won't run into some attribute of `cli` that is for some reason descendant of `Base` - but it's not an entity we should instantiate `cli` for. I don't know if any such exists and more so, I don't know if any will exist in the future. I understand your point about repetition. Please consider what I just wrote and if you don't change your opinion, I can test your version.

Reply:
@ogajduse